provider/balance_checker.go

```diff
@@ -113,7 +211,58 @@ loop:
 			balanceCheckResult = runner.Do(func() runner.Result {
 				return runner.NewResult(bc.doCheck(ctx))
 			})
+		case evt := <-subscriber.Events():
+			switch ev := evt.(type) {
+			case event.LeaseAddFundsMonitor:
+				bc.leases[ev.LeaseID] = nil
+				if bc.cfg.LeaseFundCheckInterval > 0 {
+					bc.runEscrowCheck(ctx, ev.LeaseID, leaseCheckCh)
```

Comment:
This seems to start a goroutine and then throw away the result. What is the utility in doing this? What side effect happens that is useful?